src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -22,11 +25,29 @@ import scala.concurrent.{ExecutionContext, Future}
 
 case class ClusterNotReadyException(googleProject: GoogleProject, clusterName: ClusterName) extends LeoException(s"Cluster ${googleProject.string}/${clusterName.string} is not ready yet, chill out and try again later", StatusCodes.EnhanceYourCalm)
 case class ProxyException(googleProject: GoogleProject, clusterName: ClusterName) extends LeoException(s"Unable to proxy connection to Jupyter notebook on ${googleProject.string}/${clusterName.string}", StatusCodes.InternalServerError)
-
+case class CacheException(token: String, googleMessage: String) extends LeoException(s"$token, $googleMessage")
 /**
   * Created by rtitle on 8/15/17.
   */
-class ProxyService(proxyConfig: ProxyConfig, dbRef: DbReference, clusterDnsCache: ActorRef)(implicit val system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends LazyLogging {
+class ProxyService(proxyConfig: ProxyConfig, gdDAO: DataprocDAO, dbRef: DbReference, clusterDnsCache: ActorRef)(implicit val system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends LazyLogging {
+
+  /* Cache for the bearer token and corresponding google user email */
+  private lazy val cachedAuth = CacheBuilder.newBuilder()
```

Comment:
Minor: let's not make this lazy, as constructing the cache might be semi-expensive, and it's better to do at service start time rather than the first time a notebook is accessed.